packages/@atjson/util/test/create-tree-test.ts

```diff
@@ -0,0 +1,1321 @@
+            },
+          ],
+          marks: [],
+        })
+      ).toMatchInlineSnapshot(`
+        {
+          "B0": [
+            "one",
+            {
+              "attributes": {},
+              "id": "B1",
+              "parents": [
+                "paragraph",
+              ],
+              "selfClosing": true,
+              "type": "line-break",
+            },
+            "two",
+            {
+              "attributes": {},
+              "id": "B2",
+              "parents": [
+                "paragraph",
+              ],
+              "selfClosing": true,
+              "type": "line-break",
+            },
+            "three",
+            {
+              "attributes": {},
+              "id": "B3",
+              "parents": [
+                "paragraph",
+              ],
+              "selfClosing": true,
+              "type": "line-break",
+            },
+            "four",
+          ],
+          "B4": [
+            "five",
+            {
+              "attributes": {},
+              "id": "B5",
+              "parents": [
+                "paragraph",
+              ],
+              "selfClosing": true,
+              "type": "line-break",
+            },
+            "six",
+            {
+              "attributes": {},
+              "id": "B6",
+              "parents": [
+                "paragraph",
+              ],
+              "selfClosing": true,
+              "type": "line-break",
+            },
+            "seven",
+          ],
+          "root": [
+            {
+              "attributes": {},
+              "id": "B0",
+              "parents": [],
+              "selfClosing": false,
+              "type": "paragraph",
+            },
+            {
+              "attributes": {},
+              "id": "B4",
+              "parents": [],
+              "selfClosing": false,
+              "type": "paragraph",
+            },
+          ],
+        }
+      `);
+    });
+
+    test("marks across blocks", () => {
+      expect(
+        createTree({
+          text: "\uFFFCone two\uFFFCthree four",
+          blocks: [
+            {
+              id: "B0",
+              type: "paragraph",
+              parents: [],
+              selfClosing: false,
+              attributes: {},
+            },
+            {
+              id: "B1",
+              type: "paragraph",
+              parents: [],
+              selfClosing: false,
+              attributes: {},
+            },
+          ],
+          marks: [
+            {
+              id: "M1",
+              type: "bold",
+              start: 5,
+              end: 14,
+              range: "[5..14)",
+              attributes: {},
+            },
+          ],
+        })
+      ).toMatchInlineSnapshot(`
+        {
+          "B0": [
+            "one ",
+            {
+              "attributes": {},
+              "end": 14,
+              "id": "B0-M1",
+              "range": "[5..14)",
```

Comment:
It makes sense to me that we're preserving the original range here but are also "splitting" mark in the tree, but is a little unintuitive. It may be worth calling this out specifically in the documentation